### PR TITLE
Add stress tests module for Event Hubs

### PR DESCRIPTION
**Pull request by liukun-msft**
# Description

Add event hubs stress tests project base on [azure-sdk-chaos](https://github.com/Azure/azure-sdk-tools/blob/main/tools/stress-cluster/chaos/README.md). 

See [readme](https://github.com/Azure/azure-sdk-for-java/blob/9173097d09f683b45820183e9d573329ec4fb8e1/sdk/eventhubs/azure-messaging-eventhubs-stress/README.md) file to know project content.

Resolve https://github.com/Azure/azure-sdk-for-java/issues/23954

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/eventhubs/azure-messaging-eventhubs-stress/pom.xml
Comment (liukun-msft) [inline sdk/eventhubs/azure-messaging-eventhubs-stress/pom.xml:8]: I am sure do we need a parent dependency here. Do we need to create a common parent dependency for stress purpose, or use existing client/perf parent?

File: sdk/eventhubs/azure-messaging-eventhubs-stress/Dockerfile
Comment (benbp) [inline sdk/eventhubs/azure-messaging-eventhubs-stress/Dockerfile:13]: It can be useful to have an `ENTRYPOINT` or `CMD` here just so the dockerfile can be run locally for image testing.

File: sdk/eventhubs/azure-messaging-eventhubs-stress/README.md
Comment (benbp) [inline sdk/eventhubs/azure-messaging-eventhubs-stress/README.md:24]: Preferably we route people towards the deploy script with `-Login` only, that way if we change the cluster location/name it won't break people's workflows. But I can also see how it's useful to document some of the underlying commands for testing access, etc.

File: sdk/eventhubs/azure-messaging-eventhubs-stress/README.md
Comment (benbp) [inline sdk/eventhubs/azure-messaging-eventhubs-stress/README.md:36]: FYI for local deployment with the script, we already override the namespace based on the the shell `$USER`.

File: sdk/eventhubs/azure-messaging-eventhubs-stress/README.md
Comment (benbp) [inline sdk/eventhubs/azure-messaging-eventhubs-stress/README.md:99]: For this section and above, do you think it's ok to link the stress readme in tools instead? That way if we change instructions we don't have to track down multiple places.

File: sdk/eventhubs/azure-messaging-eventhubs-stress/README.md
Comment (benbp) [inline sdk/eventhubs/azure-messaging-eventhubs-stress/README.md:119]: Can we add this step as part of the Dockerfile?

File: sdk/eventhubs/azure-messaging-eventhubs-stress/README.md
Comment (benbp) [inline sdk/eventhubs/azure-messaging-eventhubs-stress/README.md:135]: Same comment here, I think it would be better to link the main docs: https://github.com/Azure/azure-sdk-tools/blob/main/tools/stress-cluster/chaos/README.md#layout

File: sdk/eventhubs/azure-messaging-eventhubs-stress/pom.xml
Comment (benbp) [reply sdk/eventhubs/azure-messaging-eventhubs-stress/pom.xml:8]: I am not sure in the java case, but for stress tests in general it is becoming a common pattern to reference parent dependencies, though you need to pass a custom docker build directory to support it (e.g. https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/messaging/azservicebus/internal/stress/Chart.yaml#L9).

File: sdk/eventhubs/azure-messaging-eventhubs-stress/stress-test-resources.bicep
Comment (benbp) [inline sdk/eventhubs/azure-messaging-eventhubs-stress/stress-test-resources.bicep:7]: It may be more defensive to call `guid()` here as well, in case the truncated names cause resource name overlaps when only the suffixes differ.

**Comment by benbp** [comment]
This is awesome! I especially appreciate all the docs you wrote. I can't comment on the java pieces but the stress config pieces look good to me.

File: sdk/eventhubs/azure-messaging-eventhubs-stress/Dockerfile
Comment (liukun-msft) [reply sdk/eventhubs/azure-messaging-eventhubs-stress/Dockerfile:13]: Sure, I will add the `ENTRYPOINT` for testing purpose.

File: sdk/eventhubs/azure-messaging-eventhubs-stress/README.md
Comment (liukun-msft) [reply sdk/eventhubs/azure-messaging-eventhubs-stress/README.md:24]: Oh, that's great. Then I can remove this part. I couldn't have the chance to see what the error is when an un-authenticated person run that command.

File: sdk/eventhubs/azure-messaging-eventhubs-stress/README.md
Comment (liukun-msft) [reply sdk/eventhubs/azure-messaging-eventhubs-stress/README.md:99]: You are right, I will link that instead. I will focus on writing down the customized or additional content here.

File: sdk/eventhubs/azure-messaging-eventhubs-stress/README.md
Comment (liukun-msft) [reply sdk/eventhubs/azure-messaging-eventhubs-stress/README.md:119]: Good ideal. I will try this way.

File: sdk/eventhubs/azure-messaging-eventhubs-stress/README.md
Comment (liukun-msft) [reply sdk/eventhubs/azure-messaging-eventhubs-stress/README.md:36]: Oh, that's why I change namespace value in `Chart.yaml`, it still deployment on the namespace of my alias. I can note down this information to avoid misunderstanding.

File: sdk/eventhubs/azure-messaging-eventhubs-stress/src/main/java/com/azure/messaging/eventhubs/stress/config/RateMeter.java
Comment (weidongxu-microsoft) [inline sdk/eventhubs/azure-messaging-eventhubs-stress/src/main/java/com/azure/messaging/eventhubs/stress/config/RateMeter.java:26]: private or private static?

File: sdk/eventhubs/azure-messaging-eventhubs-stress/README.md
Comment (liukun-msft) [reply sdk/eventhubs/azure-messaging-eventhubs-stress/README.md:135]: For layout, I am wondering if chaos changes the layout or has some updates, there may be some gaps for project layout until we update this project accordingly. Can we keep this part to reflect current project's layout? I can add the additional link to the main docs for reference.

Comment (liukun-msft) [reply sdk/eventhubs/azure-messaging-eventhubs-stress/README.md:135]: For layout, I am wondering if chaos changes the layout or has some updates, there may be some gaps for project layout until we update this project accordingly. Can we keep this part to reflect current project's layout? I can add the additional link to the main docs for reference.

File: sdk/eventhubs/azure-messaging-eventhubs-stress/src/main/java/com/azure/messaging/eventhubs/stress/scenarios/ProcessEvents.java
Comment (weidongxu-microsoft) [inline sdk/eventhubs/azure-messaging-eventhubs-stress/src/main/java/com/azure/messaging/eventhubs/stress/scenarios/ProcessEvents.java:37]: Any intention for stdout (instead of logging)?

Or is this only a rarely used bean?

File: sdk/eventhubs/azure-messaging-eventhubs-stress/src/main/java/com/azure/messaging/eventhubs/stress/scenarios/ProcessEventsWithOptions.java
Comment (weidongxu-microsoft) [inline sdk/eventhubs/azure-messaging-eventhubs-stress/src/main/java/com/azure/messaging/eventhubs/stress/scenarios/ProcessEventsWithOptions.java:68]: Any reason for `ArrayList` instead of `List`, in type?

File: sdk/eventhubs/azure-messaging-eventhubs-stress/src/main/java/com/azure/messaging/eventhubs/stress/scenarios/ProcessEventsWithOptions.java
Comment (weidongxu-microsoft) [inline sdk/eventhubs/azure-messaging-eventhubs-stress/src/main/java/com/azure/messaging/eventhubs/stress/scenarios/ProcessEventsWithOptions.java:70]: define 40 as some constant?

File: sdk/eventhubs/azure-messaging-eventhubs-stress/src/main/java/com/azure/messaging/eventhubs/stress/scenarios/ProcessEventsWithOptions.java
Comment (weidongxu-microsoft) [inline sdk/eventhubs/azure-messaging-eventhubs-stress/src/main/java/com/azure/messaging/eventhubs/stress/scenarios/ProcessEventsWithOptions.java:86]: I assume perf would use `System.nanoTime` (as monotonic clock), instead of `currentTimeMillis` as wall clock.

File: sdk/eventhubs/azure-messaging-eventhubs-stress/README.md
Comment (benbp) [reply sdk/eventhubs/azure-messaging-eventhubs-stress/README.md:24]: It looks something like this for me:

```
$ ./eng/common/scripts/stress-testing/deploy-stress-tests.ps1 -Login
Using module powershell-yaml with version 0.4.2.
Logging in to subscription, cluster and container registry
Listing AKS cluster in Azure SDK Developer Playground/rg-stress-cluster-test

==> az aks list -g rg-stress-cluster-test --subscription Azure SDK Developer Playground -o json

ERROR: AADSTS70043: The refresh token has expired or is invalid due to sign-in frequency checks by conditional access. The token was issued on 2022-05-25T21:53:38.2830291Z and the maximum allowed lifetime for this request is 43200.
Trace ID: 55e2ee84-b016-4dd3-b612-2bc94cf21000
Correlation ID: bc61b330-123d-4471-8bf1-e35a513bb967
Timestamp: 2022-05-26 17:56:58Z
To re-authenticate, please run:
az login --scope https://management.core.windows.net//.default
Run: /home/ben/sdk/azure-sdk-tools/eng/common/scripts/stress-testing/stress-test-deployment-lib.ps1:24
Line |
  24 |      run @args
     |      ~~~~~~~~~
     | Command 'az aks list -g rg-stress-cluster-test --subscription Azure SDK Developer Playground -o json' failed with code: 1
```

File: sdk/eventhubs/azure-messaging-eventhubs-stress/README.md
Comment (benbp) [reply sdk/eventhubs/azure-messaging-eventhubs-stress/README.md:36]: Yeah it looks like we don't log anything when we're selecting the namespace and repository overrides with username. I'll add a log statement which might help in some cases.

File: sdk/eventhubs/azure-messaging-eventhubs-stress/README.md
Comment (benbp) [reply sdk/eventhubs/azure-messaging-eventhubs-stress/README.md:135]: Ah I see, that makes sense. Maybe both? Add a link to the canonical docs at the top of the section while also outlining the docs on the current java layout.

File: sdk/eventhubs/azure-messaging-eventhubs-stress/src/main/java/com/azure/messaging/eventhubs/stress/config/RateMeter.java
Comment (liukun-msft) [reply sdk/eventhubs/azure-messaging-eventhubs-stress/src/main/java/com/azure/messaging/eventhubs/stress/config/RateMeter.java:26]: yes, should be private static

File: sdk/eventhubs/azure-messaging-eventhubs-stress/src/main/java/com/azure/messaging/eventhubs/stress/scenarios/ProcessEvents.java
Comment (liukun-msft) [reply sdk/eventhubs/azure-messaging-eventhubs-stress/src/main/java/com/azure/messaging/eventhubs/stress/scenarios/ProcessEvents.java:37]: Oh, I need to change it to logger. As this test was written without log4j dependency before, I forgot to change this part.

File: sdk/eventhubs/azure-messaging-eventhubs-stress/src/main/java/com/azure/messaging/eventhubs/stress/scenarios/ProcessEventsWithOptions.java
Comment (liukun-msft) [reply sdk/eventhubs/azure-messaging-eventhubs-stress/src/main/java/com/azure/messaging/eventhubs/stress/scenarios/ProcessEventsWithOptions.java:68]: Oh, I was copied from long running test directly. I'll change it part.

File: sdk/eventhubs/azure-messaging-eventhubs-stress/src/main/java/com/azure/messaging/eventhubs/stress/scenarios/ProcessEventsWithOptions.java
Comment (liukun-msft) [reply sdk/eventhubs/azure-messaging-eventhubs-stress/src/main/java/com/azure/messaging/eventhubs/stress/scenarios/ProcessEventsWithOptions.java:70]: sure, and will also do for 64.

File: sdk/eventhubs/azure-messaging-eventhubs-stress/src/main/java/com/azure/messaging/eventhubs/stress/config/RateMeterConfig.java
Comment (saragluna) [inline sdk/eventhubs/azure-messaging-eventhubs-stress/src/main/java/com/azure/messaging/eventhubs/stress/config/RateMeterConfig.java:24]: nit: the method name will be taken as the bean name, so it's better to use a noun here.

File: sdk/eventhubs/azure-messaging-eventhubs-stress/src/main/java/com/azure/messaging/eventhubs/stress/scenarios/ProcessEventsWithOptions.java
Comment (liukun-msft) [reply sdk/eventhubs/azure-messaging-eventhubs-stress/src/main/java/com/azure/messaging/eventhubs/stress/scenarios/ProcessEventsWithOptions.java:86]: you are right. I'll change according to it.

File: sdk/eventhubs/azure-messaging-eventhubs-stress/src/main/java/com/azure/messaging/eventhubs/stress/config/RateMeterConfig.java
Comment (liukun-msft) [reply sdk/eventhubs/azure-messaging-eventhubs-stress/src/main/java/com/azure/messaging/eventhubs/stress/config/RateMeterConfig.java:24]: Good suggestion. I will use rateMeter() instead.

File: sdk/eventhubs/azure-messaging-eventhubs-stress/README.md
Comment (liukun-msft) [reply sdk/eventhubs/azure-messaging-eventhubs-stress/README.md:135]: Sure, I agree to do this way.

File: sdk/eventhubs/azure-messaging-eventhubs-stress/stress-test-resources.bicep
Comment (liukun-msft) [reply sdk/eventhubs/azure-messaging-eventhubs-stress/stress-test-resources.bicep:7]: I find resource group id is formed by subscription id and resource group name. The unique identifier is resource group name. I think testers need to prevent using the same prefix when they use scenarios values (current I remove `values.yaml` since the content of `job.yaml` are different in our test cases). But I can make a note for this potential issue.

**Review by weidongxu-microsoft — APPROVED** [review]
LGTM

Please also get approval from EH/SB (preferrable with EngSys as well).

File: sdk/eventhubs/azure-messaging-eventhubs-stress/stress-test-resources.bicep
Comment (liukun-msft) [reply sdk/eventhubs/azure-messaging-eventhubs-stress/stress-test-resources.bicep:7]: Sorry, I misunderstand before. You mean we can use guid() to generate a guid. I'll add this into name.

File: sdk/eventhubs/azure-messaging-eventhubs-stress/src/main/java/com/azure/messaging/eventhubs/stress/util/ScenarioOptions.java
Comment (conniey) [inline sdk/eventhubs/azure-messaging-eventhubs-stress/src/main/java/com/azure/messaging/eventhubs/stress/util/ScenarioOptions.java:19]: One powerful thing in Spring is configuring variables. I saw many instances of `options.get`. Maybe we can use this feature so it can be easily refactored and followed:
https://docs.spring.io/spring-boot/docs/current/reference/html/features.html#features.external-config

in this case, you would have `ScenarioOptions.getConsumerGroup()` rather than ScenarioOptions.get("CONSUMER_GROUP")`

File: sdk/eventhubs/pom.xml
Comment (conniey) [inline sdk/eventhubs/pom.xml:65]: nit: alphabetise

**Review by conniey — COMMENTED** [review]
This looks great! Just  one comment. :)

File: eng/versioning/version_client.txt
Comment (liukun-msft) [reply eng/versioning/version_client.txt:109]: Oh, right. I was previously thought about the order of creation. Changed.

File: sdk/eventhubs/azure-messaging-eventhubs-stress/src/main/java/com/azure/messaging/eventhubs/stress/util/ScenarioOptions.java
Comment (liukun-msft) [reply sdk/eventhubs/azure-messaging-eventhubs-stress/src/main/java/com/azure/messaging/eventhubs/stress/util/ScenarioOptions.java]: Sure, I changed to use spring inject all environment/arguments variables. 
Before I was struggling whether to involve spring framework as external dependency. But since it was already used for configuring some beans like `RateMeter`, it is good to highly leverage spring to configure for this project.

File: sdk/eventhubs/azure-messaging-eventhubs-stress/pom.xml
Comment (liukun-msft) [reply sdk/eventhubs/azure-messaging-eventhubs-stress/pom.xml:14]: Currently I use existing `azure-client-sdk-parent` as parent here.  If we want to have a common parent for stress test to control some specific dependencies/plugins, we can create a `azure-stress-test-parent` to do that.

**Review by conniey — APPROVED** [review]
LGTM! Maybe @anuchandy can also take a look when he is back?

File: sdk/eventhubs/azure-messaging-eventhubs-stress/Chart.yaml
Comment (ckairen) [inline sdk/eventhubs/azure-messaging-eventhubs-stress/Chart.yaml:13]: The latest version for stress-test-addons is now 0.1.20, I'd recommend upgrading to a that version

**Review by ckairen — APPROVED** [review]

File: sdk/eventhubs/azure-messaging-eventhubs-stress/Chart.yaml
Comment (liukun-msft) [reply sdk/eventhubs/azure-messaging-eventhubs-stress/Chart.yaml:13]: Sure, I'll update it in a separate PR.

**Review by anuchandy — APPROVED** [review]
thanks, Kun. LGTM! Like discussed, once pr is merged, let's sync with new test cases in the container-LRT.

**Pull request merged**
